test/freenet/node/PeerLocationTest.java

```diff
@@ -0,0 +1,110 @@
+package freenet.node;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import junit.framework.TestCase;
+
+public class PeerLocationTest extends TestCase {
+
+    private static final double[][] PEER_LOCATIONS = new double[][] {
+        // Must be sorted!
+        { 0.1 },
+        { 0.0 },
+        { 0.9 },
+        { 0.1, 0.3 },
+        { 0.1, 0.3, 0.4 },
+        { 0.0, 0.1, 0.2, 0.3, 0.3, 0.35, 0.5, 0.5, 0.99 },
+        { 0.1, 0.11, 0.12, 0.13, 0.14, 0.15, 0.16, 0.17, 0.18, 0.19, 0.20, 0.21, 0.22, 0.23, 0.24 }
+    };
```

Comment:
Is there anything specific you'd like to see tested?
From my point of view, the choice of peer locations and target locations should already capture all cases.